src/aap_eda/core/models/activation.py

```diff
@@ -98,153 +88,88 @@ class Meta:
 
     name = models.TextField(null=False, default="")
     status = models.TextField(
         choices=ActivationStatus.choices(),
         default=ActivationStatus.PENDING,
     )
     git_hash = models.TextField(null=False, default="")
-    activation = models.ForeignKey("Activation", on_delete=models.CASCADE)
+
+    # Source and activation will be completely different models
+    # Since for now are only two, two foreign keys are enough
+    # If more are added, a generic relation might be considered
+    # In this way the relation is clear and simple without the
+    # tradeoffs of a generic relation
+    activation = models.ForeignKey(
+        "Activation",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="activation_processes",
+    )
+    source = models.ForeignKey(
+        "Source",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="source_processes",
+    )
     started_at = models.DateTimeField(auto_now_add=True, null=False)
     updated_at = models.DateTimeField(null=True)
     ended_at = models.DateTimeField(null=True)
     activation_pod_id = models.TextField(null=True)
     status_message = models.TextField(null=True, default=None)
     log_read_at = models.DateTimeField(null=True)
 
-    def save(self, *args, **kwargs):
-        # when creating
-        if self._state.adding:
-            if self.status_message is None:
-                self.status_message = self._get_default_status_message()
+    class Meta:
+        db_table = "core_activation_instance"
+        ordering = ("-started_at",)
 
-                # populate latest_instance of activation at creation
-                self.activation.latest_instance = self
-        else:
-            if not bool(kwargs) or "update_fields" not in kwargs:
-                raise UpdateFieldsRequiredError(
-                    "update_fields is required to use when saving "
-                    "due to race conditions"
-                )
-            else:
-                if "status" in kwargs["update_fields"]:
-                    self._is_valid_status()
+    def __str__(self):
+        return f"{self.name} ({self.id})"
 
-            if (
-                "status_message" in kwargs["update_fields"]
-                and "status" not in kwargs["update_fields"]
-            ):
-                raise StatusRequiredError(
-                    "status_message cannot be set by itself, "
-                    "it requires status and status_message together"
+    def save(self, *args, **kwargs):
+        # populate latest_instance when creating
+        if self._state.adding:
+            parent = getattr(self, "activation", None) or getattr(
+                self,
+                "source",
+                None,
+            )
+            if not parent:
+                raise ValueError(
+                    "ActivationInstance must have either activation or source"
```

Comment:
Change this. Parent can be a property and the integrity is ensured by ProcessParentValitatorMixin